dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/ExcIEEEST4B.mo

```diff
@@ -0,0 +1,172 @@
+  parameter Types.VoltageModulePu VbMaxPu "Maximum excitation voltage (> 0).  Typical value = 11,63";
+  parameter Types.VoltageModulePu VmMaxPu "Maximum inner loop output (> VmMin).  Typical value = 99";
+  parameter Types.VoltageModulePu VmMinPu "Minimum inner loop output (< VmMax).  Typical value = -99";
+  parameter Types.VoltageModulePu VrMaxPu "Maximum voltage regulator output (> 0).  Typical value = 1";
+  parameter Types.VoltageModulePu VrMinPu "Minimum voltage regulator output (< 0).  Typical value = -0,87";
+  parameter Types.PerUnit Xl "Reactance associated with potential source (>= 0).  Typical value = 0,124";
+
+  //Input variables
+  Modelica.Blocks.Interfaces.RealInput IfdPu(start = Ifd0Pu) "Current in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-160, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {60, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+  Modelica.Blocks.Interfaces.RealInput itPuIm(start = it0Pu.im) annotation(
+    Placement(visible = true, transformation(origin = {-286, -160}, extent = {{-20, 20}, {20, -20}}, rotation = 0), iconTransformation(origin = {-40, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+  Modelica.Blocks.Interfaces.RealInput itPuRe(start = it0Pu.re) annotation(
+    Placement(visible = true, transformation(origin = {-286, -140}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-80, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+  Modelica.Blocks.Interfaces.RealInput UsPu(start = Us0Pu) annotation(
+    Placement(visible = true, transformation(origin = {-286, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 0}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput UpssPu(start = 0) "PowerSystemStabilizers output voltage in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-286, -80}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, -60}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealInput URefPu(start = Us0Pu) "Control voltage in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(origin = {-286, 80}, extent = {{-20, -20}, {20, 20}}, rotation = 0), iconTransformation(origin = {-120, 60}, extent = {{-20, -20}, {20, 20}}, rotation = 0)));
+  Modelica.ComplexBlocks.Interfaces.ComplexInput utPu(re(start = ut0Pu.re), im(start = ut0Pu.im)) "Complex voltage" annotation(
+    Placement(visible = true, transformation(origin = {-150, -60}, extent = {{-10, -10}, {10, 10}}, rotation = 0), iconTransformation(origin = {0, -120}, extent = {{-20, -20}, {20, 20}}, rotation = 90)));
+
+  //Output variable
+  Modelica.Blocks.Interfaces.RealOutput EfdPu(start = Efd0Pu) "Voltage output in pu (base UNom)" annotation(
```

Comment:
Excitation voltage in pu

base UNom or user-selected base voltage ?

Reply:
How does the user select a base?

Reply:
The user can select the parameter ExcitationPuType (or something like this).